frontend/src/views/instructors/editable-instructors-list.js

```diff
@@ -29,7 +35,15 @@ function EditableCell(props) {
 }
 
 function EditableInstructorsList(props) {
-    const { upsertInstructor, ...rest } = props;
+    const {
+        upsertInstructor,
+        inDeleteMode,
+        setInDeleteMode,
+        setDeleteDialogVisible,
+        setInstructorToDelete,
```

Comment:
Let's have `EditableInstructorsList` handle the deleting with internal state. Pass in a `deleteInstructor` prop, but all the `set*Visible...` stuff will be internal to the component.

Reply:
Need a bit of clarification here: when you say "...will be internal to the component", do you mean `Editable-Instructors-List`? Because I cannot put some of the visibility state manipulations in there since the `delete-instructor-dialog` component is inside the `index.js` view, and so the state needs to be visible from `index.js` in order to be passed into the `delete-instructor-dialog`. Could you elaborate a bit what you mean here?

Reply:
put the `DeleteInstructorDialog` inside this component. Wrap everything in a `React.Fragment`